packages/livekit-rtc/src/audio_resampler.ts

```diff
@@ -0,0 +1,166 @@
+  VERY_HIGH = SoxQualityRecipe.SOXR_QUALITY_VERYHIGH,
+}
+
+/**
+ * AudioResampler provides functionality to resample audio data from an input sample rate to
+ * an output sample rate using the Sox resampling library. It supports multiple channels and
+ * configurable resampling quality.
+ */
+export class AudioResampler {
+  #inputRate: number;
+  #outputRate: number;
+  #channels: number;
+  #ffiHandle: FfiHandle;
+
+  /**
+   * Initializes a new AudioResampler.
+   *
+   * @param inputRate - The sample rate of the input audio data (in Hz).
+   * @param outputRate - The desired sample rate of the output audio data (in Hz).
+   * @param channels - The number of audio channels (e.g., 1 for mono, 2 for stereo). Defaults to 1.
+   * @param quality - The quality setting for the resampler. Defaults to
+   * `AudioResamplerQuality.MEDIUM`.
+   */
+  constructor(
+    inputRate: number,
+    outputRate: number,
+    channels = 1,
+    quality = AudioResamplerQuality.MEDIUM,
+  ) {
+    this.#inputRate = inputRate;
+    this.#outputRate = outputRate;
+    this.#channels = channels;
+
+    const req = new NewSoxResamplerRequest({
+      inputRate,
+      outputRate,
+      numChannels: channels,
+      qualityRecipe: quality as unknown as SoxQualityRecipe,
+      inputDataType: SoxResamplerDataType.SOXR_DATATYPE_INT16I,
+      outputDataType: SoxResamplerDataType.SOXR_DATATYPE_INT16I,
+      flags: 0,
+    });
+
+    const res = FfiClient.instance.request<NewSoxResamplerResponse>({
+      message: {
+        case: 'newSoxResampler',
+        value: req,
+      },
+    });
+
+    if (res.error) {
+      throw new Error(res.error);
+    }
+
+    this.#ffiHandle = new FfiHandle(res.resampler.handle.id);
+  }
+
+  /**
+   * Push audio data into the resampler and retrieve any available resampled data.
+   *
+   * This method accepts audio data, resamples it according to the configured input and output rates,
+   * and returns any resampled data that is available after processing the input.
+   *
+   * @param data - The audio frame to resample
+   *
+   * @returns A list of {@link AudioFrame} objects containing the resampled audio data. The list may
+   * be empty if no output data is available yet.
+   */
+  push(data: AudioFrame): AudioFrame[] {
+    const req = new PushSoxResamplerRequest({
+      resamplerHandle: this.#ffiHandle.handle,
+      dataPtr: data.protoInfo().dataPtr,
+      size: data.data.length,
+    });
+
+    const res = FfiClient.instance.request<PushSoxResamplerResponse>({
+      message: {
+        case: 'pushSoxResampler',
+        value: req,
+      },
+    });
+
+    if (res.error) {
+      throw new Error(res.error);
+    }
+
+    if (res.outputPtr) {
+      return [];
+    }
+
+    const outputData = FfiClient.instance.copyBuffer(res.outputPtr, res.size);
+    return [
+      new AudioFrame(
+        new Int16Array(outputData.subarray()),
+        this.#outputRate,
+        this.#channels,
+        outputData.length,
```

Comment:
This doesn't seems right, samples per channel != bytes count 
You should divide by 2 and num_channels

Reply:
looking at our own example this seems to be fine:

https://github.com/livekit/node-sdks/blob/main/examples/publish-wav/index.ts#L60

```ts
     Math.trunc(frameSize / channels),

Reply:
I if you check above, data size was already divided by two